src/presentation/http/router/noteSettings.test.ts

```diff
@@ -281,4 +281,76 @@ describe('NoteSettings API', () => {
 
     test.todo('Return 403 when user authorized, but not member of the team');
   });
+
+  describe('PATCH /note-settings/:notePublicId/invitation-hash ', () => {
+    test('Returns status 401 when the user is not authorized', async () => {
+      const correctID = 'Pq1T9vc23Q';
+
+      const response = await global.api?.fakeRequest({
+        method: 'PATCH',
+        url: `/note-settings/${correctID}/invitation-hash`,
+      });
+
+      expect(response?.statusCode).toBe(401);
+
+      expect(response?.json()).toStrictEqual({ message: 'You must be authenticated to access this resource' });
+    });
+
+    test('Generate invitation hash by public id with 200 status, user is creator of the note', async () => {
+      const userId = 2;
+      const accessToken = global.auth(userId);
+
+      const userNote = notes.find(newNote => {
+        return newNote.creator_id === userId;
+      });
+
+      const response = await global.api?.fakeRequest({
+        method: 'PATCH',
+        headers: {
+          authorization: `Bearer ${accessToken}`,
+        },
+        url: `/note-settings/${userNote!.public_id}/invitation-hash`,
+      });
+
+      expect(response?.statusCode).toBe(200);
+
+      expect(response?.json().invitationHash).not.toEqual('FfAwyaR80C');
+
+      expect(response?.json().invitationHash).toHaveLength(10);
+    });
+
+    test('Returns status 406 when the public id does not exist', async () => {
```

Comment:
Why 406 and not 404?

Reply:
Because userInTeam policy throws this error. It can not check permission (because note in not resolved) and send 406